packages/react-i18n-webpack-plugin/src/react-i18n-webpack-plugin.ts

```diff
@@ -0,0 +1,264 @@
+import path from 'path';
+
+import webpack from 'webpack';
+import {camelCase} from 'change-case';
+import VirtualModulesPlugin from 'webpack-virtual-modules';
+import ParserHelpers from 'webpack/lib/ParserHelpers';
+import {CallExpression, Expression, SpreadElement} from 'estree';
+
+import {FindI18nImportPlugin} from './find-i18n-imports-plugin';
+import {generateID} from './utilities';
+
+const PLUGIN_NAME = 'ReactI18nPlugin';
+const TRANSLATION_DIRECTORY_NAME = 'translations';
+
+export interface Options {
+  fallbackLocale: string;
+}
+
+export class ReactI18nPlugin {
+  private options: Options;
+
+  private defaultOptions: Options = {
+    fallbackLocale: 'en',
+  };
+
+  constructor(options: Partial<Options> = {}) {
+    this.options = {
+      ...this.defaultOptions,
+      ...options,
+    };
+  }
+
+  apply(compiler: webpack.Compiler) {
+    const virtualModules = new VirtualModulesPlugin();
+    virtualModules.apply(compiler);
+
+    compiler.hooks.compilation.tap(
+      PLUGIN_NAME,
+      (
+        compilation: webpack.compilation.Compilation,
+        {
+          normalModuleFactory,
+        }: {normalModuleFactory: webpack.compilation.NormalModuleFactory},
+      ) => {
+        const handler = (parser: any) => {
+          new FindI18nImportPlugin().apply(parser);
+
+          // replace useI18n & withI18n call arguments
+          parser.hooks.evaluate
```

Comment:
So to be clear, https://webpack.js.org/api/parser/#call doesn't work here?

Reply:
It does not. This tap wont trigger. I really wish it did!